### ISSUE TITLE
Point selection highlights do not respect canvas size limits

### ISSUE THREAD

**Issue by andreasmarnold** [napari/napari#6682]
### 🐛 Bug Report

Setting points canvas size limits works fine. However, the point selection highlights are unchanged.
![points-highlight-issue](https://github.com/napari/napari/assets/29686369/b5ead4f2-9bd7-4f30-b3b3-01b785d48505)

### 💡 Steps to Reproduce

```
import napari
import numpy as np

image = np.random.random((100, 100))
points = np.array([[50, 50]])

viewer = napari.Viewer()
viewer.add_image(image)
pts_layer = viewer.add_points(points, size=100, canvas_size_limits=(5, 10), face_color='blue')

pts_layer.selected_data = {0}
pts_layer.refresh()

napari.run()
```

### 💡 Expected Behavior

I expected the selection highlights to be the same size as the points.

### 🌎 Environment

napari: 0.4.18
Platform: Windows-10-10.0.19045-SP0
Python: 3.10.13 | packaged by conda-forge | (main, Oct 26 2023, 18:01:37) [MSC v.1935 64 bit (AMD64)]
Qt: 5.15.8
PyQt5: 5.15.9
NumPy: 1.26.0
SciPy: 1.11.3
Dask: 2023.11.0
VisPy: 0.12.2
magicgui: 0.8.0
superqt: 0.6.1
in-n-out: 0.1.9
app-model: 0.2.2
npe2: 0.7.3

OpenGL:
- GL version: 4.6.0 NVIDIA 516.40
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1200, scale 1.0
- screen 2: resolution 1920x1200, scale 1.0

Settings path:
- C:\Users\arnoldam\AppData\Local\napari\napari-env_c8fb36f02872a7d738a6aed04102ebfd9d7d2515\settings.yaml

**Comment by psobolewskiPhD** [comment]
I can reproduce this on 0.4.19 and main.
Worse yet I can interact (drag) with points outside of the highlight area, so something else is bugged too.

**Comment by aeisenbarth** [comment]
Possibly related to this, I noticed that click targets mismatch and make it hard to move a point without accidentially adding new point. When zoomed out, the visible point is at the lower canvas limit size (larger than data size). When dragging the point not perfectly from the center, it does not move.

This would be explained by this bug if point selection highlights correspond to click targets.

**Comment by jni** [comment]
I think that's the opposite problem @aeisenbarth — the point selection is defined by the point "data" size, *not* the canvas size, when you are way zoomed out, selecting the point is harder. (There may of course *also* be a bug in the target detection, but I do think that is unrelated to this one.) This is made worse by macOS's implementation of OpenGL which has an absurdly small maximum point size (regardless of what canvas size limits says) — so sometimes napari thinks that a point is on top of another point, but there is no visual indication at all that this is the case. I don't really know how to fix this though, because I don't know how to introspect what the OpenGL maximum point size is...

**Comment by psobolewskiPhD** [comment]
There are a couple things at play:
1. the highlight not "sticking" to the point marker when `canvas_size_limits` is changed -- this issue, fixed by #6853 
2. the highlight (including, crucially, *thickness*) re-sizes differently than the marker, so at certain zoom levels and highlight thicknesses it's harder/impossible to tell what is selected. (this seems to be worse on highDPI screens) I think this may be a different bug.

I've not experienced selection issues when zoomed out, but when zoomed in. Those may be due to the macOS limit, but I'm not sure -- I need to do some tests with linux size by side.
